### PR TITLE
Make SignTool not fail when a cross generated file doesn't have PKT

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: #2159 

SignTool currently fails with _"Could not load file or assembly 'xyz.dll' or one of its dependencies. An attempt was made to load a program with an incorrect format."_ when a cross generated file doesn't have PKT. See sample build here.

This PR make SN not fail in such scenario and also add a new test case to check that.

### CODE REVIEW

File: src/Microsoft.DotNet.SignTool/src/Configuration.cs [hunk 1]
```diff
@@ -373,11 +373,8 @@ private static PEInfo GetPEInfo(string fullPath)
             }
 
             bool isCrossgened = ContentUtil.IsCrossgened(fullPath);
+            string publicKeyToken = ContentUtil.GetPublicKeyToken(fullPath);
```
Comment (tmat) [inline src/Microsoft.DotNet.SignTool/src/Configuration.cs]: I already mentioned this to @chcosta - we should use MetadataReader to read the public key token.

File: src/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs [hunk 2]
```diff
@@ -1219,5 +1219,18 @@ public void MissingCertificateNameButExtensionIsIgnored(string extension)
 
             Assert.False(task.Log.HasLoggedErrors);
         }
+
+        [Fact]
+        public void CrossGenneratedLibraryWithoutPKT()
```
Comment (tmat) [inline src/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs]: Typo in `Gennerated`

File: src/Microsoft.DotNet.SignTool/src/Configuration.cs (hunk 1 above)
Comment (chcosta) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: https://github.com/dotnet/arcade/pull/2079#discussion_r260932552

Comment (JohnTortugo) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: It's taking more time than I expected to get this right, so I created a new issue to address this refactoring: https://github.com/dotnet/arcade/issues/2177

**Pull request merged**